torchtnt/data/dataloaders.py

```diff
@@ -8,22 +8,22 @@
 from __future__ import annotations
 
 import logging
-from typing import Any, Dict, Iterable, Iterator, Optional, Type, TYPE_CHECKING, Union
+from typing import Any, Dict, Iterable, Iterator, Optional, Type, Union
+
+from torch.utils.data import DataLoader, Dataset
 
 from torchtnt.data.iterators import (
     DataIterationStrategy,
     DataIterationStrategyRegistry,
     MultiIterator,
 )
-
-if TYPE_CHECKING:
-    from torch.utils.data import DataLoader
+from torchtnt.data.samplers import StatefulDistributedSampler
 
 logger: logging.Logger = logging.getLogger(__name__)
 
 
-class MultiDataloader:
-    """MultiDataloader cycles through individual dataloaders passed to it.
+class MultiDataLoader:
+    """MultiDataLoader cycles through individual dataloaders passed to it.
```

Comment:
same here, if you wanted to split out this change into a separate PR, that should be easy to merge and simplify the rest of the stateful additions here